src/app/modules/main/communities/module.nim

```diff
@@ -907,6 +907,11 @@ proc applyPermissionResponse*(self: Module, communityId: string, permissions: Ta
     var permissionSatisfied = true
 
     for index, tokenCriteriaItem in tokenPermissionItem.getTokenCriteria().getItems():
+      let criteriaMet = criteriaResult.criteria[index]
+
+      if tokenCriteriaItem.criteriaMet == criteriaMet:
```

Comment:
Same question here; I believe the fix should go to `TokenPermissionsModel.updateItem` to update only those members that have actually changed, and then signal the `dataChanged` for those roles only